httpx/_types.py

```diff
@@ -45,7 +45,11 @@
 
 CertTypes = Union[str, Tuple[str, str], Tuple[str, str, str]]
 VerifyTypes = Union[str, bool, ssl.SSLContext]
-TimeoutTypes = Union[None, float, Tuple[float, float, float, float], "Timeout"]
+TimeoutTypes = Union[
+    Optional[float],
+    Tuple[Optional[float], Optional[float], Optional[float], Optional[float]],
```

Comment:
`Timeout` seems to require [the first two elements of the tuple to be not `None`](https://github.com/encode/httpx/blob/master/httpx/_config.py#L229-L230), do we want to keep those as not optional?

Reply:
It seems actually we _do_ allow those to be `None` (which makes sense - don't think there's a reason we'd want to prevent people from disabling connect/read timeouts):

https://github.com/encode/httpx/blob/440b5ab95faf94f56472e08c8c5dbf59b2175fc6/httpx/_config.py#L210-L211

(However for some reason we allow any of a 2-, 3- or 4- tuple. 🤔  Also we don't document the tuple usage in [Fine-tuning timeouts](https://www.python-httpx.org/advanced/#fine-tuning-the-configuration), instead hinting people to use kwargs, eg `Timeout(read_timeout=...)`. Is that for Requests compatibility purposes?)

Reply:
Ah sorry, I misread that 👍 

I don't think there's a lot of benefit from the tuple form FWIW.